tests/DCPS/UnitTests/DynamicData/DynamicData.cpp

```diff
@@ -582,77 +883,291 @@ TEST(Mutable, ReadValueFromUnion)
+    msg.copy((const char*)int64_union, sizeof(int64_union));
+    XTypes::DynamicData data(&msg, xcdr2, dt);
+
+    ACE_CDR::LongLong int_64;
+    DDS::ReturnCode_t ret = data.get_int64_value(int_64, 7);
+    EXPECT_EQ(ret, DDS::RETCODE_OK);
+    EXPECT_EQ(ACE_CDR::LongLong(254), int_64);
+  }
+  {
+    unsigned char uint64_union[] = {
+      0x00,0x00,0x00,0x07, // +4=4 discriminator
+      0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff// +8=12 uint_64
+    };
+
+    msg.reset();
+    msg.copy((const char*)uint64_union, sizeof(uint64_union));
+    XTypes::DynamicData data(&msg, xcdr2, dt);
+
+    ACE_CDR::ULongLong uint_64;
+    DDS::ReturnCode_t ret = data.get_uint64_value(uint_64, 8);
+    EXPECT_EQ(ret, DDS::RETCODE_OK);
+    EXPECT_EQ(ACE_CDR::ULongLong(255), uint_64);
+  }
+  {
+    unsigned char float32_union[] = {
+      0x00,0x00,0x00,0x08, // +4=4 discriminator
+      0x3f,0x80,0x00,0x00 // +4=8 float_32
+    };
+
+    msg.reset();
+    msg.copy((const char*)float32_union, sizeof(float32_union));
+    XTypes::DynamicData data(&msg, xcdr2, dt);
+
+    ACE_CDR::Float float_32;
+    DDS::ReturnCode_t ret = data.get_float32_value(float_32, 9);
+    EXPECT_EQ(ret, DDS::RETCODE_OK);
+    EXPECT_EQ(ACE_CDR::Float(1.0f), float_32);
+  }
+  {
+    unsigned char float64_union[] = {
+      0x00,0x00,0x00,0x09, // +4=4 discriminator
+      0x3f,0xf0,0x00,0x00,0x00,0x00,0x00,0x00 // +8=12 float_64
+    };
+
+    msg.reset();
+    msg.copy((const char*)float64_union, sizeof(float64_union));
     XTypes::DynamicData data(&msg, xcdr2, dt);
 
-    ACE_CDR::Boolean bool_;
-    DDS::ReturnCode_t ret = data.get_boolean_value(bool_, 15);
+    ACE_CDR::Double float_64;
+    DDS::ReturnCode_t ret = data.get_float64_value(float_64, 10);
     EXPECT_EQ(ret, DDS::RETCODE_OK);
-    EXPECT_EQ(true, bool_);
-    ret = data.get_boolean_value(bool_, 10);
-    EXPECT_EQ(ret, DDS::RETCODE_ERROR);
+    EXPECT_EQ(ACE_CDR::Double(1.0), float_64);
   }
   {
-    unsigned char str_union[] = {
-      0x00,0x00,0x00,0x14, // +4=4 dheader
-      0x20,0x00,0x00,0x00, 0x00,0x00,0x00,0x0f, // +8=12 discriminator
-      0x30,0x00,0x00,0x10, 0x00,0x00,0x00,0x04,'a','b','c','\0' // +12=24 str
+    unsigned char float128_union[] = {
+      0x00,0x00,0x00,0x0a, // +4=4 discriminator
+      0x3f,0xff,0,0,0,0,0,0,0,0,0,0,0,0,0,0     // 16=20 float_128
     };
+
     msg.reset();
-    msg.copy((const char*)str_union, sizeof(str_union));
+    msg.copy((const char*)float128_union, sizeof(float128_union));
     XTypes::DynamicData data(&msg, xcdr2, dt);
 
-    ACE_CDR::Char* str = 0;
-    DDS::ReturnCode_t ret = data.get_string_value(str, 16);
+    ACE_CDR::LongDouble float_128;
+    DDS::ReturnCode_t ret = data.get_float128_value(float_128, 11);
     EXPECT_EQ(ret, DDS::RETCODE_OK);
-    EXPECT_STREQ("abc", str);
-    CORBA::string_free(str);
-
-    str = 0;
-    ret = data.get_string_value(str, 10);
-    EXPECT_EQ(ret, DDS::RETCODE_ERROR);
+    ACE_CDR::LongDouble expected = 1.0L;
```

Comment:
```suggestion
set_float128_value(expected);
```
There is a CI build that failed to do the assignment from `1.0L` to `ACE_CDR::LongDouble`. Using `set_float128_value` should fix it. You can use this function in other similar assignments to `ACE_CDR::LongDouble`.